# HG changeset patch
# User Ben Abbott  <bpabbott@mac.com>
# Date 1335838754 14400
#      Mon Apr 30 22:19:14 2012 -0400
# Branch stable
# Node ID 89504d0a5c5b069c0add097cf7bb0b270e185734
# Parent  000cd393f3c1391f57569bb79746cd5c2314807f
poissrnd should return zero when lambda = 0. (Bug # 36326)

poissrnd.m: poissonrnd should return zero when lambda = 0.
Add demos.

diff --git a/scripts/statistics/distributions/poissrnd.m b/scripts/statistics/distributions/poissrnd.m
--- a/scripts/statistics/distributions/poissrnd.m
+++ b/scripts/statistics/distributions/poissrnd.m
@@ -71,34 +71,33 @@ function rnd = poissrnd (lambda, varargi
 
   if (isa (lambda, "single"))
     cls = "single";
   else
     cls = "double";
   endif
 
   if (isscalar (lambda))
-    if (lambda > 0 && lambda < Inf)
+    if (lambda >= 0 && lambda < Inf)
       rnd = randp (lambda, sz);
       if (strcmp (cls, "single"))
         rnd = single (rnd);
       endif
     else
       rnd = NaN (sz, cls);
     endif
   else
     rnd = NaN (sz, cls);
 
-    k = (lambda > 0) & (lambda < Inf);
+    k = (lambda >= 0) & (lambda < Inf);
     rnd(k) = randp (lambda(k));
   endif
 
 endfunction
 
-
 %!assert(size (poissrnd (2)), [1, 1]);
 %!assert(size (poissrnd (ones(2,1))), [2, 1]);
 %!assert(size (poissrnd (ones(2,2))), [2, 2]);
 %!assert(size (poissrnd (1, 3)), [3, 3]);
 %!assert(size (poissrnd (1, [4 1])), [4, 1]);
 %!assert(size (poissrnd (1, 4, 1)), [4, 1]);
 
 %% Test class of input preserved
@@ -113,8 +112,11 @@ endfunction
 %!error poissrnd (1, 2, ones(2))
 %!error poissrnd (i)
 %!error poissrnd (1, 2, -1)
 %!error poissrnd (1, [2 -1 2])
 %!error poissrnd (ones(2,2), 3)
 %!error poissrnd (ones(2,2), [3, 2])
 %!error poissrnd (ones(2,2), 2, 3)
 
+%!assert (poissrnd (0, 1, 1), 0)
+%!assert (poissrnd ([0, 0, 0], [1, 3]), [0 0 0])
+
